src/pkcs11/framework-pkcs15.c

```diff
@@ -553,6 +553,27 @@ public_key_created(struct pkcs15_fw_data *fw_data, const struct sc_pkcs15_id *id
 	return SC_ERROR_OBJECT_NOT_FOUND;
 }
 
+static void
+pkcs15_cert_extract_label(struct pkcs15_cert_object *cert)
+{
+	sc_log(context, "pkcs15_cert_extract_label() called. Current label: %s", cert->cert_p15obj->label);
+
+	/* if we didn't get a label, set one based on the CN */
+	if (*cert->cert_p15obj->label == '\0') { /* can't be NULL -- static array */
+		static const struct sc_object_id cn_oid = {{ 2, 5, 4, 3, -1 }};
+		u8 *cn_name = NULL;
+		size_t cn_len = 0;
+		int rv = sc_pkcs15_get_name_from_dn(context,
+			cert->cert_data->subject, cert->cert_data->subject_len,
+			&cn_oid, &cn_name, &cn_len);
+		sc_log(context, "pkcs15_cert_extract_label(): Name from DN is %s", cn_name);
+		if (rv == SC_SUCCESS) {
+			memcpy(cert->cert_p15obj->label, cn_name, cn_len);
```

Comment:
Thanks for comment. Fixed by truncating the length to `SC_PKCS15_MAX_LABEL_SIZE-1` in case it is longer.